hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/EvictionOperator.java

```diff
@@ -58,18 +60,22 @@ boolean isEvictionEnabled() {
         return evictionStrategy != null && evictionPolicyEvaluator != null;
     }
 
-    boolean evictIfRequired() {
-        if (isEvictionEnabled()) {
-            return evictionStrategy.evict(cache, evictionPolicyEvaluator, evictionChecker, listener);
+    void evictIfRequired() {
+        if (!isEvictionEnabled()) {
+            return;
+        }
+
+        for (int i = 0; evictionChecker.isEvictionRequired() && i < MAX_EVICTION_ATTEMPTS; i++) {
+            // we already established we should evict -> we can pass EVICT_ALWAYS to the eviction strategy
+            evictionStrategy.evict(cache, evictionPolicyEvaluator, EVICT_ALWAYS, listener);
```

Comment:
Just checking - this is a best-effort mechanism where we evict up to `MAX_EVICTION_ATTEMPTS` - even if the cache capacity has been exceeded more than that limit?

Reply:
yes. you want to spread the cost of cleaning it up across multiple calls. to prevent bad latency spikes.

Reply:
I added comment to the property itself